### PR TITLE
chore: update yarn to version 4.5.3

### PR DESCRIPTION
**Pull request by ksted**
## Why:

Our Yarn version is quite outdated by now. The new version is supposed to provide several enhancements.

[Yarn 4 Release Notes](https://yarnpkg.com/blog/release/4.0)

Furthermore, until now we've had to maintain the Yarn version separately here and in other repositories. By moving the setup process to a script, we can keep a single source of truth throughout.

## Describe your changes

- Updates Yarn to version 4.5.3.
- Removes plugins, they're built into Yarn 4.
- Moves Yarn setup to a script. 
- Moves NPM auth token to environment.
- Updates the setup instructions in the README.

## Checklist before requesting a review

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

### CODE REVIEW

File: yarn.ps1 [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  Write-Output "Usage: yarn.ps1 [option...]"
+  Write-Output ""
+  Write-Output "   -h, --help       Show this help message"
+  Write-Output "   -v, --version    Show script version"
+  Write-Output "   -f, --force      Force delete .yarnrc.yml without asking"
+  Write-Output ""
+}
+
+function Show-Version {
+  Write-Output "Script version 1.0"
```
Comment (maxiroellplenty) [inline yarn.ps1]:
```suggestion
  Write-Output "1.0"
```

File: yarn.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  echo "Usage: $0 [option...]"
+  echo
+  echo "   -h, --help       Show this help message"
+  echo "   -v, --version    Show script version"
+  echo "   -f, --force      Force delete .yarnrc.yml without asking"
+  echo
+}
+
+function display_version() {
+  echo "Script version 1.0"
```
Comment (maxiroellplenty) [inline yarn.sh]:
```suggestion
  echo "1.0"
```

File: package.json [hunk 3]
```diff
@@ -9,7 +9,9 @@
   "scripts": {
     "prepare": "is-ci || husky install",
     "update:sdk": "yarn up @plentymarkets/shop-api",
-    "setup": "yarn && yarn build && yarn dev",
+    "setup": "if [ \"$OS\" = \"Windows_NT\" ]; then yarn setup:windows; else yarn setup:unix; fi",
```
Comment (maxiroellplenty) [inline package.json]: setup does not work if there is a .yarnrc.yml file with a different yarn version. We should use either use the force flag or have a some other information how to fix that.

File: yarn.sh [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  if [ -z "$NPM_AUTH_TOKEN" ]; then
+    echo "NPM_AUTH_TOKEN is not set in the GitHub Actions environment"
+    exit 1
+  fi
+  yarn config set npmScopes.plentymarkets.npmAuthToken $NPM_AUTH_TOKEN
+elif grep -q "NPM_AUTH_TOKEN" ./apps/web/.env; then
+  export $(grep "NPM_AUTH_TOKEN" ./apps/web/.env | xargs)
+  yarn config set npmScopes.plentymarkets.npmAuthToken $NPM_AUTH_TOKEN
+else
+  echo "NPM_AUTH_TOKEN not found in ./apps/web/.env"
```
Comment (maxiroellplenty) [inline yarn.sh]: NPM_AUTH_TOKEN  should be added to the .env.example

File: yarn.sh [hunk 5]
```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
```
Comment (maxiroellplenty) [inline yarn.sh:2]: you can setup some colors: 
``` sh
# Colors
DARKGRAY='\033[1;30m'
RED='\033[0;31m'
LIGHTRED='\033[1;31m'
GREEN='\033[0;32m'
YELLOW='\033[1;33m'
BLUE='\033[0;34m'
PURPLE='\033[0;35m'
LIGHTPURPLE='\033[1;35m'
CYAN='\033[0;36m'
WHITE='\033[1;37m'
SET='\033[0m'

```

and then use them like this:
``` sh  
echo -e "${YELLOW}No base branch found for repository ${repo_path} ${SET}"
 ``` 
start by using echo -e and then follow with your color ${YELLOW} and reset the color back to default by ending with ${SET}

 might be good for the error messages like the missing npm token.

File: package.json (hunk 3 above)
Comment (maxiroellplenty) [reply package.json]: So I had to run ./yarn.sh -f to install update yarn.
Once that completed successfully, I ran into this issue when running the `yarn setup` again
Bildschirmfoto 2024-12-04 um 10 49 17
I guess after running yarn.sh i dont need the setup step. Still the setup command fails for me. Not sure if its becuase of zsh or something.

Comment (maxiroellplenty) [reply package.json]: and yes I did add the NPM_AUTH_TOKEN to the .env after running the yarn.sh and did run it again with no issues. just the yarn setup fails for me

Comment (ksted) [reply package.json]: I can't replicate your issue here. For me it runs fine every time. Also I think we shouldn't use the force flag here, the command should prompt you for confirmation instead.

I've made one small change, can you try again?

File: yarn.sh [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+delete_yarnrc
+yarn set version stable
+yarn set version 4.5.3 --yarn-path
+yarn config set nodeLinker node-modules
+yarn config set npmScopes.plentymarkets.npmRegistryServer "https://npm.pkg.github.com"
+yarn config set npmScopes.plentymarkets.npmAlwaysAuth true
+
+if [ -n "$GITHUB_ACTIONS" ]; then
+  if [ -z "$NPM_AUTH_TOKEN" ]; then
+    echo -e "${RED}NPM_AUTH_TOKEN is not set in the GitHub Actions environment ${SET}"
```
Comment (maxiroellplenty) [inline yarn.sh:69]: also need to declare the `SET` var right?

File: package.json (hunk 3 above)
Comment (maxiroellplenty) [reply package.json]: ➜  plentyshop-pwa git:(chore/yarn-version) yarn setup
Unbound variable "OS"
command not found: then
command not found: else
command not found: fi

Comment (ksted) [reply package.json]: What if you run `yarn setup:unix`? Does the actual command work as expected?

**Review by maxiroellplenty — APPROVED** [review]

**Pull request merged**